src/components/SearchBar/index.jsx

```diff
@@ -22,21 +23,36 @@ const DesktopSearchBarContainer = styled.div`
   `};
 `;
 
-function SearchBar() {
-  const [location, setLocation] = useState('');
-  const [status, useStatus] = useState(false);
-  const [title, setTitle] = useState('');
+function SearchBar({ onSearch }) {
+  const [location, setLocation] = useState("");
+  const [status, setStatus] = useState(false);
+  const [title, setTitle] = useState("");
 
   // @TODO Implement search and filter functionality.
-  const handleOnChangeByLocation = () => {};
-  const handleOnChangeByStatus = () => {};
-  const handleOnChangeByTitle = () => {};
-  const handleOnSearch = () => {};
+  const handleOnChangeByLocation = (e) => setLocation(e.target.value);
+  const handleOnChangeByStatus = () => setStatus(!status);
+  const handleOnChangeByTitle = (e) => setTitle(e.target.value);
+  const handleOnSearch = () => {
+    const filtered = jobsData.filter((job) => {
+      const jobPosition = job.position ? job.position.toLowerCase() : "";
+      const jobLocation = job.location ? job.location.toLowerCase() : "";
+      const matchesTitle = title
+        ? jobPosition.includes(title.toLowerCase())
+        : true;
+      const matchesLocation = location
+        ? jobLocation.includes(location.toLowerCase())
+        : true;
+      return matchesTitle && matchesLocation;
+    });
+    onSearch(filtered);
+  };
 
   return (
     <>
```

Comment:
I noticed while testing that I am unable to search when I press "Enter" on my keyboard. Can you wrap the `SearchBar` component with a `form` element so we can search with the keyboard input as well? Something like this:

```suggestion
    <form onSubmit={handleOnSearch()}>